### DISCUSSION TITLE
Python

### DISCUSSION THREAD

**Discussion by 23Monikayadav** [PyMySQL/mysqlclient#669] [Q&A] [answered]
I am newest on GitHub.
And I will try to learn python..
I don't know how to use GitHub for learning and making projects

**Comment by abhishek887229** [comment, marked as answer]
Hello Monika Yadav,
you took a good step ahead with your decision to learn Python, to Learning Python and using git hub are the two sides of a coin, to learn Python, you can take references from youtube and online tutorials, to learn use the approach **learn by doing** , you can learn git hub very easily here are some video that worked for me so it will help you too I guess.
here are the links:-

1.  [python](https://youtu.be/YYXdXT2l-Gg?feature=shared)
2. [git and github ](https://youtu.be/Ez8F0nW6S-w?feature=shared)

**Reply by 23Monikayadav** [reply]
Thank you sir ✨

**Comment by Abcdzkg** [comment]
Hello! GitHub is a very useful tool for developers to store their code and collaborate with others. Here are some ways you can use GitHub to learn Python and create projects:

1.Fork and clone repositories: GitHub allows you to fork someone else's repository to your own GitHub account, which means you can make a copy of the code. You can then clone the repository to your local computer using Git, allowing you to modify and explore the codebase. This is a great way to learn new coding techniques and see how other developers have approached problems.
2.Create your own repositories: You can create your own repositories on GitHub to store your own code. This allows you to organize your projects, track changes, and collaborate with others. As you learn Python, you can create repositories for your practice code, small projects, or even larger projects you're working on.
3.Join GitHub communities: GitHub has a large community of developers, and you can join these communities to learn new skills, ask questions, and find collaborators for your projects. For example, you can join the Python community on GitHub, which has a lot of resources and helpful people.
4.Use GitHub Pages: GitHub Pages allows you to host websites directly from your GitHub repository. You can create a website to showcase your Python projects or even use it as a portfolio to show potential employers what you've built.
5.Collaborate with others: GitHub allows you to collaborate with others on projects. You can find someone who's interested in working with you on a Python project and start collaborating. This allows you to learn from others, share knowledge, and build something together.
Remember that learning Python and using GitHub takes time, so don't be overwhelmed if it seems challenging at first. Practice and experiment with different things, and you'll gradually get better at it.

**Comment by methane** [comment]
This discussion forum is about PyMySQL, not about Python.

**Comment by zhishiheibao** [comment]
Python has a vast ecosystem of libraries. Learn how to use popular ones such as NumPy for numerical operations, Pandas for data manipulation, and Matplotlib or Seaborn for data visualization.
Depending on your domain, explore domain-specific libraries like TensorFlow or PyTorch for machine learning, Flask or Django for web development, and requests for working with APIs.

**Comment by ZeitWang** [comment]
I am fucking accompanying with you
